amaranth/vendor/_siliconblue.py

```diff
@@ -382,21 +382,21 @@ def create_missing_domain(self, name):
                                          i_CLKHFPU=1,
                                          p_CLKHF_DIV=f"0b{self.hfosc_div:02b}",
                                          o_CLKHF=clk_i)
-                delay = int(100e-6 * self.default_clk_frequency)
+                delay = Period(us=100) // self.default_clk_period
             # Internal low-speed clock: 10 KHz.
             elif self.default_clk == "SB_LFOSC":
                 clk_i = Signal()
                 m.submodules += Instance("SB_LFOSC",
                                          i_CLKLFEN=1,
                                          i_CLKLFPU=1,
                                          o_CLKLF=clk_i)
-                delay = int(100e-6 * self.default_clk_frequency)
+                delay = Period(us=100) // self.default_clk_period
```

Comment:
while this is not a problem introduced by this PR, this is dangerously wrong: for safety reasons, this should be obviously rounded *up*, not down. for `SB_LFOSC` in particular, we are one ulp away from rounding this 0.9999 to a 0, instead of a 1.

... actually, I believe this code should be rounding up and also adding an extra margin of `1` cycle because there is no guarantee when the first rising edge actually comes up.